Add changelog note for CodeQL v2.10.5 deprecation (#1993)
github/codeql-action: CHANGELOG.md

## [UNRELEASED]

> - Add a deprecation warning for customers using CodeQL version 2.11.5 and earlier. These versions of CodeQL were discontinued on 8 November 2023 alongside GitHub Enterprise Server 3.7, and will be unsupported by the next minor release of the CodeQL Action. [#1993](https://github.com/github/codeql-action/pull/1993)

Reviewer: Use North American style dates, I think.

```suggestion
- Add a deprecation warning for customers using CodeQL version 2.11.5 and earlier. These versions of CodeQL were discontinued on November, 8 2023 alongside GitHub Enterprise Server 3.7, and will be unsupported by the next minor release of the CodeQL Action, 2.23. [#1993](https://github.com/github/codeql-action/pull/1993)
```

Reviewer: Nevermind, I take that back. We're using British dates elsewhere in this changelog, so lets just keep it here.

But, would still be nice to explicitly use the version of the action where it is deprecated.

```suggestion
- Add a deprecation warning for customers using CodeQL version 2.11.5 and earlier. These versions of CodeQL were discontinued on 8 November 2023 alongside GitHub Enterprise Server 3.7, and will be unsupported by the next minor release of the CodeQL Action, 2.23. [#1993](https://github.com/github/codeql-action/pull/1993)
```

Author: ```suggestion
- Add a deprecation warning for customers using CodeQL version 2.11.5 and earlier. These versions of CodeQL were discontinued on 8 November 2023 alongside GitHub Enterprise Server 3.7, and will be unsupported by CodeQL Action v2.23.0 and later. [#1993](https://github.com/github/codeql-action/pull/1993)
```
